Code/Framework/AzToolsFramework/Tests/Prefab/PrefabFocus/PrefabFocusTests.cpp

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Component/EntityId.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <AzToolsFramework/Entity/PrefabEditorEntityOwnershipInterface.h>
+#include <AzToolsFramework/Prefab/PrefabFocusInterface.h>
+#include <Prefab/PrefabTestFixture.h>
+
+namespace UnitTest
+{
+    class PrefabFocusTests
+        : public PrefabTestFixture
+    {
+    protected:
+        void GenerateTestHierarchy()
+        {
+            /*
+             *  City (Prefab Container)
+             *  |_  City
+             *  |_  Street (Prefab Container)
+             *      |_  Car (Prefab Container)
+             *      |   |_ Passenger
+             *      |_  SportsCar (Prefab Container)
+             *          |_ Passenger
+             */
+
+            m_entityMap["passenger1"] = CreateEntity("Passenger1");
+            m_entityMap["passenger2"] = CreateEntity("Passenger2");
+            m_entityMap["city"] = CreateEntity("City");
+
+            AzToolsFramework::EditorEntityContextRequestBus::Broadcast(
+                &AzToolsFramework::EditorEntityContextRequests::HandleEntitiesAdded,
+                AzToolsFramework::EntityList{ m_entityMap["passenger1"], m_entityMap["passenger2"], m_entityMap["city"] });
+
+            AZStd::unique_ptr<AzToolsFramework::Prefab::Instance> carInstance =
+                m_prefabSystemComponent->CreatePrefab({ m_entityMap["passenger1"] }, {}, "test/car");
+            ASSERT_TRUE(carInstance);
+            m_instanceMap["car"] = carInstance.get();
```

Comment:
suggestion: Some guide comments here might be helpful for those less familiar with prefab setup (also having a helper to wrap this call might be nice as it repeats a few times below just with different values)